app/models/article/node.rb

```diff
@@ -1,24 +1,32 @@
-module Article::Node
-  class Base
-    include Cms::Model::Node
-
-    default_scope ->{ where(route: /^article\//) }
-  end
-
-  class Page
-    include Cms::Model::Node
-    include Cms::Addon::NodeSetting
-    include Cms::Addon::Meta
-    include Event::Addon::PageList
-    include Category::Addon::Setting
-    include Cms::Addon::TagSetting
-    include Cms::Addon::OpendataRef::Site
-    include Cms::Addon::Release
-    include Cms::Addon::DefaultReleasePlan
-    include Cms::Addon::MaxFileSizeSetting
-    include Cms::Addon::GroupPermission
-    include History::Addon::Backup
-
-    default_scope ->{ where(route: "article/page") }
-  end
-end
+module Article::Node
+  class Base
+    include Cms::Model::Node
+
+    default_scope ->{ where(route: /^article\//) }
+  end
+
+  class Page
+    include Cms::Model::Node
+    include Cms::Addon::NodeSetting
+    include Cms::Addon::Meta
+    SS::Site.each do |s|
+      if s.facebook_access_token.present?
+        include Cms::Addon::NodeAutoPostSetting
+      elsif s.twitter_consumer_key.present? && s.twitter_consumer_secret.present? \
+          && s.twitter_access_token.present? && s.twitter_access_token_secret.present?
+        include Cms::Addon::NodeAutoPostSetting
+      end
+    end
+    include Event::Addon::PageList
+    include Category::Addon::Setting
+    include Cms::Addon::TagSetting
+    include Cms::Addon::OpendataRef::Site
+    include Cms::Addon::Release
+    include Cms::Addon::DefaultReleasePlan
+    include Cms::Addon::MaxFileSizeSetting
+    include Cms::Addon::GroupPermission
+    include History::Addon::Backup
+
+    default_scope ->{ where(route: "article/page") }

```

Comment:
__[rails_best_practices]__
* default_scope is evil [:ref](http://rails-bestpractices.com/posts/2013/06/15/default_scope-is-evil/)

<!-- Comment by [SideCI](https://sideci.com) -->